library/src/scala/quoted/Constant.scala

```diff
@@ -0,0 +1,7 @@
+package scala.quoted
+
+import scala.runtime.quoted.Runner
+
+object Constant {
+  def unapply[T](expr: Expr[T])(implicit runner: Runner[T]): Option[T] = runner.toConstantOpt(expr)
+}
```

Comment:
We should leave design space for other extractors as well. Some thoughts:

- it's better all extractors are located in the same file (as there is not many)
- the name `runner` doesn't make sense to programmers.
    - FYI, in gestalt it's called `Toolbox`, in macros it's called `Universe`.
- `ConstantExpr` can be removed, and Liftables could be implemented with an implicit `Runner`

Reply:
Needing an implicit `Runner` is something that should be avoided. That would create an unnecessary dependency on the compiler/toolbox. In the current design, if you only quote, splice and inline the user never needs to add the runner.

Reply:
Thanks for the explanation. I just want to point out another logically consistent design. As long as the design tradeoffs are clear,  I think it's fine to optimise.

Reply:
In essence, the runner is equivalent to the toolbox except that we only use it to interpret expression at runtime (compile and run, show and deconstruction). The constructor are the quotes themselves, `'(if (~cond) ~e1 else ~e2)` is the constructor for an if expression. The only missing constructor is for literal constants from runtime primitive values, which we special case in `ConstantExpr`. It is true that having `ConstantExpr` is more performant but it is not an optimization. Thought we could decide to compile `'(5)` to a `ConstantExpr` to avoid the pickling memory and runtime costs.